### PR TITLE
Support for `||` (OR) operator without using it for concatenation

### PR DESCRIPTION
**Pull request by adkhare**
Contributing for https://github.com/youtube/vitess/issues/2466

### CODE REVIEW

**Comment by adkhare** [comment]
@bbeaudreault `||` operator which should operate just like `OR` operator

**Comment by adkhare** [comment]
@sougou Please can you review the same.

File: go/vt/sqlparser/token.go [hunk 1]
```diff
@@ -334,8 +334,16 @@ func (tkn *Tokenizer) Scan() (int, []byte) {
...
 		case eofChar:
 			return 0, nil
-		case '=', ',', ';', '(', ')', '+', '*', '%', '&', '|', '^', '~':
+		case '=', ',', ';', '(', ')', '+', '*', '%', '&', '^', '~':
 			return int(ch), nil
+		case '|':
+			switch tkn.lastChar {
+			case '|':
+				tkn.next()
+				return OR_PIPE, nil
```
Comment (sougou) [inline go/vt/sqlparser/token.go]: You could just return `OR` here instead.

**Comment by bbeaudreault** [comment]
thanks for helping @adkhare!

**Comment by adkhare** [comment]
@sougou just completed implementing your recommendations. tests passed as well

**Comment by sougou** [comment]
Does #2502 supersede this one?

**Comment by adkhare** [comment]
Yes. Closing this as the parent ticket takes care of this

**Pull request closed without merging**